drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -405,6 +423,7 @@ static int nrf5_tx(const struct device *dev,
 		return -EIO;
 	}
 
+	k_sem_take(&nrf5_radio->tx_started, K_FOREVER);
```

Comment:
I'm not sure about blocking on a semaphore here. It is possible that there is a requested transmission in long future (like 30 seconds). It would block possibility to transit other frames meantime. But I don't have any better ideas right. I think we can leave it as it is, possibly refactor later.

Reply:
It would also break the rule that the `tx` operation is blocking. Maybe the `net_pkt_set_timestamp` API should be changed to something else (e.g. `radio->tx_scheduled(..)`)?